src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -1187,7 +1187,7 @@ void SpatialPooler::inhibitColumnsLocal_(vector<Real>& overlaps, Real density,
       }
     }
 
-    if (numBigger < numActive) {
+    if (overlaps[column] > 0 && numBigger < numActive) {
```

Comment:
Yes. Similar to the Python, though, it'd make sense to do this check sooner and not compute `numBigger` and `numActive`.